src/azure-cli/azure/cli/command_modules/role/tests/latest/test_graph.py

```diff
@@ -132,43 +132,57 @@ def test_application_set_scenario(self):
                 }
             ])
         })
-
+        app_id = None
         # create app through general option
         self.cmd('ad app create --display-name {name} --homepage {app} --identifier-uris {app}',
                  checks=self.check('identifierUris[0]', '{app}'))
 
         # set app password
         result = self.cmd('ad app credential reset --id {app} --append --password "test" --years 2').get_output_in_json()
-        self.assertTrue(result['appId'])
+        app_id = result['appId']
+        self.assertTrue(app_id)
 
-        # show/list app
-        self.cmd('ad app show --id {app}',
-                 checks=self.check('identifierUris[0]', '{app}'))
-        self.cmd('ad app list --display-name {name}', checks=[
-            self.check('[0].identifierUris[0]', '{app}'),
-            self.check('length([*])', 1)
-        ])
+        try:
```

Comment:
I am just curious about why we need this `try/finally`. It seems that if we meet any error during the try, this code will not fail and just delete the app.

Reply:
Will the `self.check` still work? For some commands which don't have `self.check`, does this `try/finally` matter?

Reply:
This is a good question.
The `try/finally` is to ensure the graph application created will always get removed even the test exits before the step of `az ad app delete`; otherwise the leftover will accumulate in the test tenant like garbages 
The test code here is not in a good shape. The right way is to create a test preparer like `ResourceGroupPreparer` which guarantees the chance to clean up anything. I will let @leol7 handle that once he ramps up
`self.check` will only be picked up if the command succeeds. If the command fails, CLI's test FX will throw and control flow will get the the `finally` block